### ISSUE TITLE
WebStorm: Exception on Save when "compile file" is checked

### ISSUE THREAD

**Issue by markiewb** [dubreuia/intellij-plugin-save-actions#110]
The option "compile file" was checked.
```
com/intellij/openapi/compiler/CompilerManager
java.lang.NoClassDefFoundError: com/intellij/openapi/compiler/CompilerManager
	at com.dubreuia.processors.CompileProcessor$1.run(CompileProcessor.java:39)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:314)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.k(IdeEventQueue.java:821)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:649)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.ClassNotFoundException: com.intellij.openapi.compiler.CompilerManager PluginClassLoader[com.dubreuia, 0.21] com.intellij.ide.plugins.cl.PluginClassLoader@1814622a
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:65)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 21 more

```

**Comment by dubreuia** [comment]
I'll just remove compile option, afaik it doesn't even work properly

**Comment by markiewb** [comment]
Yes, remove it for products which don't provide this action/class

Den 2 okt. 2017 1:34 em skrev "Alexandre DuBreuil" <notifications@github.com

**Comment by markiewb** [comment]
The compile action works - at least for Java code.  But only for the
current file. Classes,  which depend on the current file/class,  are not
compiled.

Den 2 okt. 2017 1:34 em skrev "Alexandre DuBreuil" <notifications@github.com

**Comment by dubreuia** [comment]
That is the intended behavior. I'll look into it soon, I'd like to add more products

**Comment by dubreuia** [comment]
Fixed in 0.23-RC2, it also deactivates build in PyCharm, I think this is intended behavior since the compiler is probably platform based. 

[save-actions_0.23-RC2.jar.zip](https://github.com/dubreuia/intellij-plugin-save-actions/files/1405269/save-actions_0.23-RC2.jar.zip)

**Comment by dubreuia** [comment]
Fixed https://github.com/dubreuia/intellij-plugin-save-actions/releases/tag/v0.23
